### PR TITLE
Install terraform cli and vs code plugin in dev container

### PR DESCRIPTION
**Pull request by wbreza**
In terraform projects we need to ensure that both the Terraform CLI and the Terraform vs code extensions are installed by default in the dev container.

### CODE REVIEW

File: templates/todo/projects/python-mongo/.repo/terraform/.devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    # Install Terraform CLI
+    && sudo apt-get update && sudo apt-get install -y gnupg software-properties-common \
+    && wget -O- https://apt.releases.hashicorp.com/gpg | \
+        gpg --dearmor | \
+        sudo tee /usr/share/keyrings/hashicorp-archive-keyring.gpg \
+    && echo "deb [signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] \
+        https://apt.releases.hashicorp.com $(lsb_release -cs) main" | \
+        sudo tee /etc/apt/sources.list.d/hashicorp.list \
+    && sudo apt update \
+    && sudo apt-get install terraform=1.1.7
```
Comment (ellismg) [inline templates/todo/projects/python-mongo/.repo/terraform/.devcontainer/Dockerfile:13]: Just wondering - do we want to enforce exactly 1.1.7 here or should we just not put a version specifier, so we always pick up the "latest" terraform?

**Review by ellismg — APPROVED** [review]

File: templates/todo/projects/python-mongo/.repo/terraform/.devcontainer/Dockerfile (hunk 1 above)
Comment (wbreza) [reply templates/todo/projects/python-mongo/.repo/terraform/.devcontainer/Dockerfile]: Had internal discussion and agree that we should take latest CLI.

File: templates/todo/projects/python-mongo/.repo/terraform/.devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    # Install Terraform CLI
+    && sudo apt-get update && sudo apt-get install -y gnupg software-properties-common \
+    && wget -O- https://apt.releases.hashicorp.com/gpg | \
+        gpg --dearmor | \
+        sudo tee /usr/share/keyrings/hashicorp-archive-keyring.gpg \
+    && echo "deb [signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] \
+        https://apt.releases.hashicorp.com $(lsb_release -cs) main" | \
+        sudo tee /etc/apt/sources.list.d/hashicorp.list \
+    && sudo apt update \
+    && sudo apt-get install terraform
```
Comment (HadwaAbdelhalem) [inline templates/todo/projects/python-mongo/.repo/terraform/.devcontainer/Dockerfile:13]: should not  we select  a certain terraform version?

Comment (wbreza) [reply templates/todo/projects/python-mongo/.repo/terraform/.devcontainer/Dockerfile:13]: We want latest so we can detect if/when a terraform updates starts to break azd templates.

**Pull request merged**